### DISCUSSION TITLE
Yup Mixed Causes Validation Error

### DISCUSSION THREAD

**Discussion by SawkaDev** [omar-dulaimi/prisma-yup-generator#13] [Q&A]
Hello

I have a simple prisma schema shown here

```
model UnitType {
  id          String  @id @default(uuid())
  length      Int
  width       Int
  height      Int?
  name        String?
  description String?
}
```

prisma-yup-generator then creates the following schema

```
export const UnitTypeCreateInputObjectSchema = Yup.object({
  id: Yup.string(),
  length: Yup.number().required(),
  width: Yup.number().required(),
  height: Yup.mixed().oneOfSchemas([Yup.number()]),
  name: Yup.mixed().oneOfSchemas([Yup.string()]),
  description: Yup.mixed().oneOfSchemas([Yup.string()]),
  Tenant: TenantCreateNestedOneWithoutUnitTypesInputObjectSchema
});
```

My question is, how come the height property is using `Yup.mixed().oneOfSchemas()` function. This is causing my form to not validate properly. Should'nt it just generate the `height` property as a normal `Yup.number()`

This is the form validation error I am getting

`{"height":{"message":"Not all items in height match one of the allowed schemas","type":"one-of-schemas","ref":{}}}`

However, in my form I am clearly entering in a number. The length and width work fine.

**Comment by omar-dulaimi** [comment]
Half of this is fixed, and testing it found a second defect that is still live. Thank you for the precise schema; I used it exactly as written.

Generated from your `UnitType` model on current code:

```ts
height:      oneOfSchemas([Yup.number()]),
name:        oneOfSchemas([Yup.string()]),
description: oneOfSchemas([Yup.string()]),
```

Note it is now a plain `oneOfSchemas([...])` call rather than `Yup.mixed().oneOfSchemas([...])`. That method was added to Yup's prototype at runtime and TypeScript was never told it existed, so every call site was a type error and the emitted files needed `// @ts-nocheck` to compile at all. It is an ordinary exported function now, and the header is gone.

**Then I ran your schema rather than reading it:**

```
ACCEPT  { length: 1, width: 2 }                    // optional absent
ACCEPT  { length: 1, width: 2, height: 3 }         // optional present
REJECT  { length: 1, width: 2, height: null }      // <- still wrong
REJECT  { length: 1, width: 2, height: 'nope' }    // correctly rejected
```

So the case you reported, a validation error on an untouched optional field, passes now. But `height Int?` in Prisma is **nullable**, and `{ height: null }` is rejected with "Not all items in height match one of the allowed schemas". That is a real bug and it is mine, not yours.

It is the same class of defect I fixed in the sibling `prisma-zod-generator` last week, where every nullable column rejected `null` because optional and nullable were being conflated. `.optional()` admits `undefined`; it does not admit `null`. The Yup generator has the same conflation in `oneOfSchemas`.

I am closing this thread since the question you asked has an answer, but the null case is tracked and will be fixed. If you are still on this generator and `null` writes matter to you, say so and I will prioritise it.
